src/main/java/org/jabref/gui/fieldeditors/contextmenu/EditorContextAction.java

```diff
@@ -0,0 +1,95 @@
+import javafx.scene.input.Clipboard;
+
+import org.jabref.gui.actions.ActionFactory;
+import org.jabref.gui.actions.SimpleCommand;
+import org.jabref.gui.actions.StandardActions;
+import org.jabref.gui.keyboard.KeyBindingRepository;
+import org.jabref.gui.util.BindingsHelper;
+import org.jabref.logic.util.OS;
+
+import com.sun.javafx.scene.control.Properties;
+
+public class EditorContextAction extends SimpleCommand {
+
+    private static final boolean SHOW_HANDLES = Properties.IS_TOUCH_SUPPORTED && !OS.OS_X;
+
+    private final StandardActions command;
+    private final TextInputControl textInputControl;
+
+    public EditorContextAction(StandardActions command, TextInputControl textInputControl) {
+        this.command = command;
+        this.textInputControl = textInputControl;
+
+        BooleanProperty editableBinding = textInputControl.editableProperty();
+        BooleanBinding hasTextBinding = Bindings.createBooleanBinding(() -> textInputControl.getLength() > 0, textInputControl.textProperty());
+        BooleanBinding hasStringInClipboardBinding = (BooleanBinding) BindingsHelper.constantOf(Clipboard.getSystemClipboard().hasString());
+        BooleanBinding hasSelectionBinding = Bindings.createBooleanBinding(() -> textInputControl.getSelection().getLength() > 0, textInputControl.selectionProperty());
+        BooleanBinding allSelectedBinding = Bindings.createBooleanBinding(() -> textInputControl.getSelection().getLength() == textInputControl.getLength());
+        BooleanBinding maskTextBinding = (BooleanBinding) BindingsHelper.constantOf(textInputControl instanceof PasswordField); // (maskText("A") != "A");
+
+        if (SHOW_HANDLES) {
+            this.executable.bind(
+                    switch (command) {
+                        case COPY -> editableBinding.and(maskTextBinding.not()).and(hasSelectionBinding);
+                        case CUT -> maskTextBinding.not().and(hasSelectionBinding);
+                        case PASTE -> editableBinding.and(hasStringInClipboardBinding);
+                        case DELETE -> editableBinding.and(hasSelectionBinding);
+                        case SELECT_ALL -> hasTextBinding.and(allSelectedBinding.not());
+                        default -> BindingsHelper.constantOf(true);
+                    });
+        } else {
+            this.executable.bind(
+                    switch (command) {
+                        case COPY -> editableBinding.and(maskTextBinding.not()).and(hasSelectionBinding);
+                        case CUT -> maskTextBinding.not().and(hasSelectionBinding);
+                        case PASTE -> editableBinding.and(hasStringInClipboardBinding);
+                        case DELETE -> editableBinding.and(hasSelectionBinding);
+                        case SELECT_ALL -> hasTextBinding.and(allSelectedBinding.not()); // why was this disabled before?
+                        default -> BindingsHelper.constantOf(true);
+                    });
+        }
```

Comment:
The SELECT_ALL case was originally disabled, if SHOW_HANDLES was false. Why?

Reply:
No idea why...